dotnet/samples/HomeAutomation/Plugins/MyLightPlugin.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.ComponentModel;
+using Microsoft.SemanticKernel;
+
+namespace HomeAutomation.Plugins;
+
+/// <summary>
+/// Class that represents a controllable light.
+/// </summary>
+[Description("Represents a light")]
+public class MyLightPlugin
+{
+    private bool _turnedOn;
+
+    public MyLightPlugin(bool turnedOn = false)
```

Comment:
Could we demonstrate DI in the light plugin constructor? E.g., logging or a room/house service?

Reply:
Sure. But are you sure you want to increase the complexity of the light plugin (and of the example) to add more DI in a way that doesn't illustrate anything related to SK?

Reply:
Yes, a lot of the questions I'm getting right now is how to use DI with plugins in Semantic Kernel. I think the value prop we have by deeply integrating with DI is a key part of why someone would want to use Semantic Kernel instead of another SDK like LangChain.

Reply:
To keep things easy and more obvious, I've created a MyAlarmPlugin which itself takes in a MyTimePlugin as a dependency to illustrate using dependencies to create other dependencies.